### DISCUSSION TITLE
Using OpenFilePickerAsync and SaveFilePickerAsync on Android

### DISCUSSION THREAD

**Discussion by Yveske1966** [AvaloniaUI/Avalonia#15066] [Q&A] [answered]
I want to use the following code:
AppTopLevel!.StorageProvider.OpenFilePickerAsync (fileOptions);
AppTopLevel!.StorageProvider.SaveFilePickerAsync (fileOptions);

With AppTopLevel getting filled in in App.axaml.cs as shown in https://github.com/AvaloniaUI/Avalonia/discussions/13599:
    public override void OnFrameworkInitializationCompleted ()
    {
        MainViewModel = new ();

        if (ApplicationLifetime is IClassicDesktopStyleApplicationLifetime desktop)
        {
            desktop.MainWindow = new MainWindow
            {
                DataContext = mainViewModel
            };

            AppTopLevel = TopLevel.GetTopLevel (desktop.MainWindow);
        }
        else if (ApplicationLifetime is ISingleViewApplicationLifetime singleViewPlatform)
        {
            singleViewPlatform.MainView = new MainView
            {
                DataContext = mainViewModel
            };

            AppTopLevel = TopLevel.GetTopLevel (singleViewPlatform.MainView);
        }

        base.OnFrameworkInitializationCompleted ();
    }

The solution works for Windows (IClassicDesktopStyleApplicationLifetime), but fails for Android (ISingleViewApplicationLifetime) because AppTopLevel equals null. I haven't tested on OSX and IOS yet.

**Comment by workgroupengineering** [comment]
See https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/ViewInteraction/MvvmDialogSample

**Comment by maxkatz6** [comment, marked as answer]
Unless you need StorageProvider in the app startup, please call `Toplevel.GetTopLevel` when it's necessary on user input.
